src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { actions, currentTodoSelector } from '../../features/currentTodo';
+import { useAppSelector } from '../../app/store';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { useDispatch } from 'react-redux';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  const currentTodo = useAppSelector(currentTodoSelector) as Todo;
```

Comment:
Ensure that `currentTodo` is not null before accessing its properties. Consider adding a check or using optional chaining to prevent runtime errors if `currentTodo` is null.